NPE in `TestConvergence`

From https://circleci.com/gh/cockroachdb/cockroach/4285:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x5a3298]

goroutine 1370 [running]:
github.com/cockroachdb/cockroach/proto.FromNetAddr(0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/proto/gossip.go:29 +0x28
github.com/cockroachdb/cockroach/gossip.(*client).gossip(0xc2081c4b40, 0xc2080830e0, 0xc208052900, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/gossip/client.go:124 +0x3d2
github.com/cockroachdb/cockroach/gossip.func·001()
	/go/src/github.com/cockroachdb/cockroach/gossip/client.go:85 +0x1a6
github.com/cockroachdb/cockroach/util.func·001()
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 1 [chan receive]:
testing.RunTests(0xa05a40, 0xd6c6c0, 0x29, 0x29, 0xc207fff301)
	/usr/src/go/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc20800c960, 0x50)
	/usr/src/go/src/testing/testing.go:485 +0x6c
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc20800c960)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x28
github.com/cockroachdb/cockroach/gossip_test.TestMain(0xc20800c960)
	/go/src/github.com/cockroachdb/cockroach/gossip/main_test.go:35 +0x28
main.main()
	github.com/cockroachdb/cockroach/gossip/_test/_testmain.go:132 +0x1d1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0xd88e80)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1038 +0x75
created by github.com/cockroachdb/cockroach/util/log.init·1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:604 +0xcc

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/src/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 339 [runnable]:
sync.(*WaitGroup).Wait(0xc208052910)
	/usr/src/go/src/sync/waitgroup.go:132 +0x169
github.com/cockroachdb/cockroach/util.(*Stopper).Stop(0xc208052900)
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:132 +0x57
github.com/cockroachdb/cockroach/gossip/simulation.(*Network).Stop(0xc2080f16c0)
	/go/src/github.com/cockroachdb/cockroach/gossip/simulation/network.go:177 +0x2c
github.com/cockroachdb/cockroach/gossip_test.verifyConvergence(0xa, 0x8, 0x5f5e100, 0xc208082120)
	/go/src/github.com/cockroachdb/cockroach/gossip/convergence_test.go:44 +0x1a5
github.com/cockroachdb/cockroach/gossip_test.TestConvergence(0xc208082120)
	/go/src/github.com/cockroachdb/cockroach/gossip/convergence_test.go:52 +0x89
testing.tRunner(0xc208082120, 0xd6ca80)
	/usr/src/go/src/testing/testing.go:447 +0xbf
created by testing.RunTests
	/usr/src/go/src/testing/testing.go:555 +0xa8b

goroutine 1557 [runnable]:
net/rpc.(*service).call(0xc2080f0a00, 0xc2080f0280, 0xc2080bbb78, 0xc208063680, 0xc2080fbde0, 0x8ca440, 0xc2089efad0, 0x16, 0x8b2520, 0xc20815f640, ...)
	/usr/src/go/src/net/rpc/server.go:376
created by net/rpc.(*Server).ServeCodec
	/usr/src/go/src/net/rpc/server.go:476 +0x44a

goroutine 1567 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1247 [IO wait]:
net.(*pollDesc).Wait(0xc20810f100, 0x72, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20810f100, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20810f0a0, 0xc20821c000, 0x1000, 0x1000, 0x0, 0x7f3db47d7d38, 0xc2081c1330)
	/usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2089671e8, 0xc20821c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc208179c20)
	/usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc208179c20, 0xc2089ee600, 0x0, 0x0)
	/usr/src/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x7f3db47d9e28, 0xc208179c20, 0x7f3db47d7bc0, 0x0, 0x0)
	/usr/src/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc2081ee380, 0x7f3db47d9df0, 0xc2081ee4f0, 0x0, 0xa054d8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).readRequestHeader(0xc2081ee380, 0xc208179c20, 0xc2081ee4f0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:180 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).ReadRequestHeader(0xc2081ee380, 0xc2089debe0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:64 +0x6e
net/rpc.(*Server).readRequestHeader(0xc2080f0200, 0x7f3db47d9db0, 0xc2081ee380, 0x0, 0x0, 0xc2089debe0, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2080f0200, 0x7f3db47d9db0, 0xc2081ee380, 0xc208063600, 0xc2089deb60, 0x8ca440, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/src/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2080f0200, 0x7f3db47d9db0, 0xc2081ee380)
	/usr/src/go/src/net/rpc/server.go:461 +0x6d
github.com/cockroachdb/cockroach/rpc.(*Server).serveConn(0xc208062080, 0x7f3db47d9b48, 0xc2089671e8, 0xc2089deb40)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:258 +0x95
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP(0xc208062080, 0x7f3db47d9cf0, 0xc2089d8fa0, 0xc208962a90)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:115 +0x47d
net/http.serverHandler.ServeHTTP(0xc208053080, 0x7f3db47d9cf0, 0xc2089d8fa0, 0xc208962a90)
	/usr/src/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2089d8640)
	/usr/src/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/src/go/src/net/http/server.go:1751 +0x35e

goroutine 1569 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1570 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1578 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1568 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1564 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1576 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 980 [IO wait]:
net.(*pollDesc).Wait(0xc208011950, 0x72, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208011950, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2080118f0, 0xc20895e000, 0x1000, 0x1000, 0x0, 0x7f3db47d7d38, 0xc2089ea708)
	/usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2089d0f10, 0xc20895e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc20819b320)
	/usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc20819b320, 0xc2089ec600, 0x0, 0x0)
	/usr/src/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x7f3db47d9e28, 0xc20819b320, 0x7f3db47d7bc0, 0x0, 0x0)
	/usr/src/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc208230fc0, 0x7f3db47d9df0, 0xc208231130, 0x0, 0xa054d8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).readRequestHeader(0xc208230fc0, 0xc20819b320, 0xc208231130, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:180 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).ReadRequestHeader(0xc208230fc0, 0xc2080e65c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:64 +0x6e
net/rpc.(*Server).readRequestHeader(0xc2080f0300, 0x7f3db47d9db0, 0xc208230fc0, 0x0, 0x0, 0xc2080e65c0, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2080f0300, 0x7f3db47d9db0, 0xc208230fc0, 0xc208063700, 0xc2081188e0, 0x8ca440, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/src/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2080f0300, 0x7f3db47d9db0, 0xc208230fc0)
	/usr/src/go/src/net/rpc/server.go:461 +0x6d
github.com/cockroachdb/cockroach/rpc.(*Server).serveConn(0xc208062500, 0x7f3db47d9b48, 0xc2089d0f10, 0xc2080e65a0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:258 +0x95
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP(0xc208062500, 0x7f3db47d9cf0, 0xc2089d8f00, 0xc2081d01a0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:115 +0x47d
net/http.serverHandler.ServeHTTP(0xc2080533e0, 0x7f3db47d9cf0, 0xc2089d8f00, 0xc2081d01a0)
	/usr/src/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2089d8b40)
	/usr/src/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/src/go/src/net/http/server.go:1751 +0x35e

goroutine 979 [IO wait]:
net.(*pollDesc).Wait(0xc2080118e0, 0x72, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2080118e0, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208011880, 0xc2089dc000, 0x1000, 0x1000, 0x0, 0x7f3db47d7d38, 0xc2089ffc80)
	/usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2089d0f00, 0xc2089dc000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc20819b2c0)
	/usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc20819b2c0, 0xc2089ee800, 0x0, 0x0)
	/usr/src/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x7f3db47d9e28, 0xc20819b2c0, 0x7f3db47d7bc0, 0x0, 0x0)
	/usr/src/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc208230e00, 0x7f3db47d9df0, 0xc208230f70, 0x0, 0xa054d8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).readRequestHeader(0xc208230e00, 0xc20819b2c0, 0xc208230f70, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:180 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).ReadRequestHeader(0xc208230e00, 0xc208118b00, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:64 +0x6e
net/rpc.(*Server).readRequestHeader(0xc2080f0300, 0x7f3db47d9db0, 0xc208230e00, 0x0, 0x0, 0xc208118b00, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2080f0300, 0x7f3db47d9db0, 0xc208230e00, 0xc208063700, 0xc2081d3100, 0x8ca440, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/src/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2080f0300, 0x7f3db47d9db0, 0xc208230e00)
	/usr/src/go/src/net/rpc/server.go:461 +0x6d
github.com/cockroachdb/cockroach/rpc.(*Server).serveConn(0xc208062500, 0x7f3db47d9b48, 0xc2089d0f00, 0xc2080e6560)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:258 +0x95
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP(0xc208062500, 0x7f3db47d9cf0, 0xc2089d8e60, 0xc2081d00d0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:115 +0x47d
net/http.serverHandler.ServeHTTP(0xc2080533e0, 0x7f3db47d9cf0, 0xc2089d8e60, 0xc2081d00d0)
	/usr/src/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2089d86e0)
	/usr/src/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/src/go/src/net/http/server.go:1751 +0x35e

goroutine 1000 [IO wait]:
net.(*pollDesc).Wait(0xc20810eed0, 0x72, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20810eed0, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20810ee70, 0xc20822a000, 0x1000, 0x1000, 0x0, 0x7f3db47d7d38, 0xc208216140)
	/usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2089d15d8, 0xc20822a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc208126360)
	/usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc208126360, 0xc2081d5000, 0x0, 0x0)
	/usr/src/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x7f3db47d9e28, 0xc208126360, 0x7f3db47d7bc0, 0x0, 0x0)
	/usr/src/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc208231340, 0x7f3db47d9df0, 0xc2082314b0, 0x0, 0xa054d8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).readRequestHeader(0xc208231340, 0xc208126360, 0xc2082314b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:180 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).ReadRequestHeader(0xc208231340, 0xc208163bc0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:64 +0x6e
net/rpc.(*Server).readRequestHeader(0xc2080f0300, 0x7f3db47d9db0, 0xc208231340, 0x0, 0x0, 0xc208163bc0, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2080f0300, 0x7f3db47d9db0, 0xc208231340, 0xc208063700, 0xc208988560, 0x8ca440, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/src/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2080f0300, 0x7f3db47d9db0, 0xc208231340)
	/usr/src/go/src/net/rpc/server.go:461 +0x6d
github.com/cockroachdb/cockroach/rpc.(*Server).serveConn(0xc208062500, 0x7f3db47d9b48, 0xc2089d15d8, 0xc208118aa0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:258 +0x95
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP(0xc208062500, 0x7f3db47d9cf0, 0xc2089d9680, 0xc2081d1790)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:115 +0x47d
net/http.serverHandler.ServeHTTP(0xc2080533e0, 0x7f3db47d9cf0, 0xc2089d9680, 0xc2081d1790)
	/usr/src/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2089d9400)
	/usr/src/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/src/go/src/net/http/server.go:1751 +0x35e

goroutine 1573 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1562 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1563 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1565 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1566 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1558 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1559 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1560 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1561 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1574 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1579 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1575 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1577 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1572 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff

goroutine 1194 [IO wait]:
net.(*pollDesc).Wait(0xc208219c60, 0x72, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208219c60, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208219c00, 0xc2089f8000, 0x1000, 0x1000, 0x0, 0x7f3db47d7d38, 0xc208112a08)
	/usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208966038, 0xc2089f8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc2081e8a20)
	/usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc2081e8a20, 0xc208122700, 0x0, 0x0)
	/usr/src/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x7f3db47d9e28, 0xc2081e8a20, 0x7f3db47d7bc0, 0x0, 0x0)
	/usr/src/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc2089d5a40, 0x7f3db47d9df0, 0xc2089d5bb0, 0x0, 0xa054d8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).readRequestHeader(0xc2089d5a40, 0xc2081e8a20, 0xc2089d5bb0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:180 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).ReadRequestHeader(0xc2089d5a40, 0xc2081637a0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:64 +0x6e
net/rpc.(*Server).readRequestHeader(0xc2080f0200, 0x7f3db47d9db0, 0xc2089d5a40, 0x0, 0x0, 0xc2081637a0, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2080f0200, 0x7f3db47d9db0, 0xc2089d5a40, 0xc208063600, 0xc208163700, 0x8ca440, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/src/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2080f0200, 0x7f3db47d9db0, 0xc2089d5a40)
	/usr/src/go/src/net/rpc/server.go:461 +0x6d
github.com/cockroachdb/cockroach/rpc.(*Server).serveConn(0xc208062080, 0x7f3db47d9b48, 0xc208966038, 0xc2081636a0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:258 +0x95
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP(0xc208062080, 0x7f3db47d9cf0, 0xc20805cdc0, 0xc2081d1ad0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:115 +0x47d
net/http.serverHandler.ServeHTTP(0xc208053080, 0x7f3db47d9cf0, 0xc20805cdc0, 0xc2081d1ad0)
	/usr/src/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc20805cd20)
	/usr/src/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/src/go/src/net/http/server.go:1751 +0x35e

goroutine 1571 [runnable]:
github.com/cockroachdb/cockroach/gossip.func·013()
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:296
created by github.com/cockroachdb/cockroach/gossip.(*infoStore).processCallbacks
	/go/src/github.com/cockroachdb/cockroach/gossip/infostore.go:300 +0x2ff
```

--- a/rpc/client.go
+++ b/rpc/client.go
@@ -172,10 +172,10 @@ func (c *Client) connect(opts *retry.Options, context *Context) error {
 			return err
 		}
 
-		break
+		return nil
 	}
 
-	return nil
+	return util.Errorf("system is stopping")
 }
 
 // IsConnected returns whether the client is connected.

